scripts/configurator/parser/cmake_var.py

```diff
@@ -1,9 +1,84 @@
-from typing import Dict, Any, Type, Mapping
+from collections import namedtuple
+from typing import Dict, Any, Optional, Type, Mapping, List
+import re
+from sys import stderr
+
+
+class metadata:
+   """By convention, first row is considered a user comment
+   and is not considered in the metadata parsing process.
+   Raises a ValueError if a keyword definition is repeated twice, or if an
+   invalid keyword is provided."""
+   keyword_expr = re.compile(r"^([A-Z_]+) (.*)")
+
+   def __init__(self, comment: List[str]) -> None:
+      kw_tuple = namedtuple("kw_tuple", ["parser", "setter", "var"])
+      """ if we wanted to use typehinting on kw_tuple, we should derivate a new
+      class from namedtuple and set arguments directly from there"""
+      self.depends_on: Optional[List[str]] = None
+      self.group: Optional[str] = None
+      self.realtype: Optional[str] = None
+
+      self.kw = {
+         "REALTYPE":
+            kw_tuple(self.parse_realtype, self.get_setter("realtype"), self.realtype),
+         "DEPENDS_ON":
+            kw_tuple(self.parse_depends_on, self.get_setter("depends_on"),self.depends_on),
+         "GROUP":
+            kw_tuple(self.parse_group, self.get_setter("group"), self.group),
+      }
+
+      for i, line in enumerate(comment):
+         match = self.keyword_expr.match(line)
+         if not match:
+            continue
+
+         key = match.group(1)
+         if i == 0:
+            if key in self.kw:
+               msg = """WARNING: First line uses a reserved keyword: {}.
+               This will be ignored."""
+               stderr.write(msg.format(key))
+            continue
+
+         elif key not in self.kw:
+            msg = "{} is not a valid metadata keyword."
+            raise ValueError(msg.format(key))
+
+         elif self.kw[key].var is not None:
```

Comment:
Do you have a test for this case? I'm not convinced that if the member variable changes in the object, the change will be reflected here, simply because python doesn't have explicit references: in kw we simply stored a reference to a particular value; after the member var is re-assigned, here we'll see the value of the old reference. That's why I suggested to have parser, setter and _getter_, where the getter is implemented similarly as the setter with `nonlocal` etc. 

You could simply test that by adding an `assert(self.kw[key].var is not None)` after calling the setter. I guess the assert will by triggered, because the var we have in `kw` will still be None.

Reply:
I will try, and correct if wrong